lobsterpy/plotting/__init__.py

```diff
@@ -226,3 +236,421 @@ def _broaden(energies: np.ndarray, population: np.ndarray, sigma=None, cutoff=4.
+            outer_key = "All"
+            key = bond_key + label_addition
+            self._cohps[outer_key].update(
+                {
+                    key: {
+                        "energies": energies,
+                        "COHP": populations,
+                        "ICOHP": int_populations,
+                        "efermi": cohps.efermi,
+                    }
+                }
+            )
+
+    def get_plot(
+        self,
+        xlim=None,
+        rangeslider=False,
+        ylim=None,
+        plot_negative=None,
+        integrated=False,
+        invert_axes=True,
+        sigma=None,
+    ):
+        """
+        Get an interactive plotly figure showing the COHPs.
+
+        Args:
+            xlim: Specifies the x-axis limits. Defaults to None for
+                automatic determination.
+            rangeslider: Adds a plotly.graph_objs.layout.xaxis.Rangeslider
+                object to figure to allow easy manipulation of x-axis..
+            ylim: Specifies the y-axis limits. Defaults to None for
+                automatic determination.
+            plot_negative: It is common to plot -COHP(E) so that the
+                sign means the same for COOPs and COHPs. Defaults to None
+                for automatic determination: If are_coops is True, this
+                will be set to False, else it will be set to True.
+            integrated: Switch to plot ICOHPs. Defaults to False.
+            invert_axes: Put the energies onto the y-axis, which is
+                common in chemistry.
+            sigma: Standard deviation of Gaussian broadening applied to
+                population data. If this is unset (None) no broadening will be
+                added.
+
+        Returns:
+            A  plotly.graph_objects.Figure object.
+        """
+        if self.are_coops and not self.are_cobis:
+            cohp_label = "COOP"
+        elif self.are_cobis and not self.are_coops:
+            cohp_label = "COBI"
+        elif self.are_cobis and self.are_coops:
+            raise ValueError(
+                "Plot data should not contain COBI and COOP data at same time"
+            )
+        else:
+            cohp_label = "COHP"
+
+        if plot_negative is None:
+            plot_negative = (not self.are_coops) and (not self.are_cobis)
+
+        if integrated:
+            cohp_label = "I" + cohp_label + " (eV)"
+
+        if plot_negative:
+            cohp_label = "\u2212" + cohp_label
+
+        if self.zero_at_efermi:
+            energy_label = "$E - E_f$ (eV)"
+        else:
+            energy_label = "$E$ (eV)"
+
+        # Setting up repeating color scheme (same as for matplotlib plots in .mplstyle)
+        palette = [
+            "#e41a1c",
+            "#377eb8",
+            "#4daf4a",
+            "#984ea3",
+            "#ff7f00",
+            "#ffff33",
+            "#a65628",
+            "#f781bf",
+            "#999999",
+        ]
+        pal_iter = cycle(palette)
```

Comment:
If matplotlib is available we can get the current colours from `matplotlib.rcParams['axes.prop_cycle'].by_key()['color']`. That would allow stylesheets etc. to be used directly.

Matplotlib is a hard dependency of Pymatgen, so should always be available.

Reply:
Hi @ajjackson , I am not much of an expert user of matplotlib. I could not understand what you meant exactly. If you can ellaborate a bit, It would help me to address this issue. Plotly I think does not support matplotlib stylesheets. Maybe I am wrong. 

Also, The interactive plotter class is not generating any matplotlib figure object.

Reply:
rcParams exposes the current default appearance settings. I believe that when styles are used the rcParams will also reflect this, so this is somewhere we can look for Matplotlib styling information.

At the moment we have a hard-coded colour-palette here so there is no way to dynamically switch to a set that might better accommodate e.g. colour-blindness, neurodivergence, a stylish dark background...

Although Plotly does not directly use Matplotlib styling, we can pull out numbers if it is available and pass them on. The colour cycle is set in a slightly complicated format as part of a broader "prop_cycle" that loops over line properties; the `by_key()['color']` part breaks out the part of interest here.

I appreciate that we don't really want to port the _entire_ matplotlibrc to equivalent Plotly settings, but in my experience end-users really like to have some control over colours for data presentation!

Reply:
Thanks @ajjackson  for the detailed explanation. So we would like to provide the option to users to change colors used in the plot. I thought of this solution which could address the hardcoded issue

1.  directly use matplotlib.rcParams['axes.prop_cycle'].by_key()['color']
2. add an agrument colormap to get_plot method in Interactive plotter and then use matplotlib to get hexcodes something like this

```
colormap = matplotlib.cm.get_cmap('tab20')
num_colors = 20
palette = colormap(range(num_colors ))
hex_palette = [matplotlib.colors.rgb2hex(color) for color in palette]
```

Let me know what you think about this @ajjackson and @JaGeo  😃

Reply:
Thanks for this suggestion. Now it should be more flexible with colors we use

Reply:
This looks much easier to customise and extend :+1:

Reply:
Thanks